content/post/review-programs-with-graphs.md

```diff
@@ -0,0 +1,322 @@
+applied, reducing the problem of many-body interacting system to a series of single-body problems and although still
+slow, it is highly valuable for many tasks in physics, chemistry and material science. Many approximation methods
+have been developed to make this more feasible by getting estimates instead of exact answers.
+
+### Neural Networks for predicting properties of molecules
+
+So in 2017, several researchers at Google Brain Residence program spent time applying recent ML optimization methods
+for predicting properties of molecules:
+
+* many [software simulators](https://en.wikipedia.org/wiki/List_of_quantum_chemistry_and_solid-state_physics_software)
+already exist for DFT, but are slow and still computationally intensive
+
+* in 2014 those simulators were used to build a [reference dataset — QM9](http://quantum-machine.org/datasets/),
+  suitable for supervised learning
+
+* a new “featurization” method was proposed, for looking at molecular structure as a graph: atoms as nodes and bonds as edges
+
+* a new variation of the Gated Graph Neural Network architecture was proposed, particularly suited for summarizing
+  properties of such graphs
+
+Every such graph representing a molecule could be treated as a “computational graph”, thus the usual Neural Network
+training techniques could be applied to build node embeddings and given such a model the desired properties of
+the molecule could be learned in a supervised fashion, as a function of the whole graph.
+
+{{% caption src="https://cdn-images-1.medium.com/max/2000/1*NGX58AUORTa-cosWIGKntg.png" title="MPNN illustration from paper Neural 'Message Passing for Quantum Chemistry'" %}}
+MPNN illustration from [https://arxiv.org/abs/1704.01212](https://arxiv.org/abs/1704.01212)
+{{% /caption %}}
+
+[“Predicting properties of molecules” blog post by Google Research](https://research.googleblog.com/2017/04/predicting-properties-of-molecules-with.html)
+dives deeper into details, but this work, in particular, has several valuable meta-lessons to teach on conducting a
+novel research in applied Machine Learning:
+
+* a single, shared benchmark QM9 was used (based on DFT, previous simulation approach)
+
+* paper#1 [“Machine learning prediction errors better than DFT accuracy”](http://"https://arxiv.org/abs/1702.05532)
+ a systematic assessment of machine learning methods on the QM9 benchmark was conducted, and a new featurization
+ method proposed
+
+* paper#2  [“Neural Message Passing for Quantum Chemistry”](https://arxiv.org/abs/1704.01212) a general model family
+ “Message Passing Neural Networks” (MPNNs) was proposed, that reformulate previous NN models invariant
+ to graph symmetries into a single common framework. Novel model in MPNN family developed, that improve results
+ ~factor of 4. High-level interpretation of result: models that can leverage inherent symmetries in data will tend
+ to generalize better
+
+## Paper highlights
```

Comment:
As much as I understand why the previous section was valuable to give context everything I was able to get from it is that this technique comes from an existing ML application for Quantum Chemistry.

Maybe that's what you should say without getting into the details of how that was developed? Or maybe you need to get into more detail so I actually understand it better.
The whole section was supposed to tell me what Gated Graph Neural Networks are, and I still have no idea after reading it twice.

Reply:
> The whole section was supposed to tell me what Gated Graph Neural Networks are
this might be a miss-understanding - the paper does not explain what that is, as noted above, there is a separate paper explaining the architecture.

> this technique comes from an existing ML application for Quantum Chemistry

yes, that is basically this + valuable context around it: DFTs, how graph structure is used, how Google Residency program systematically supported this research - all information collected though some "detective" work by looking around.

I have shown this blog post draft to multiple other reviewers and the feedback was that this is the most interesting part of the post that reveals hidden context/motivation behind the research, so I would prefer keeping.